classes/task/crawl_task.php

```diff
@@ -31,29 +30,55 @@
 
 /**
  * crawl_task
+ * Creates a batch of crawls to be done in an ad hoc task.
+ * This gives us control over how parallel it is, and also when the workload is processed (eg at night).
  *
  * @package    tool_crawler
  * @copyright  2016 Brendan Heywood <brendan@catalyst-au.net>
  * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
  */
 class crawl_task extends \core\task\scheduled_task {
-
     /**
      * Get task name
      */
     public function get_name() {
-        return get_string('pluginname', 'tool_crawler');
+        return get_string('crawl_task', 'tool_crawler');
     }
 
     /**
      * Execute task
      */
     public function execute() {
-        if (\tool_crawler\robot\crawler::get_config()->disablebot === '1') {
+        $config = \tool_crawler\robot\crawler::get_config();
+        if ($config->disablebot === '1') {
             return;
         }
-
-        tool_crawler_crawl();
+        $limitadhoctasks = $config->crawl_task;
```

Comment:
'crawl_task' seems like a poor name for this

Reply:
Have changed the name to what it is: max workers